range.go

```diff
@@ -197,3 +197,28 @@ func (m rangeOpMatcher) String() string {
 		m.eqScanOp.String(),
 	)
 }
+
+// IndexFromConditions returns the name of the index or the base table to use, along with the key info
+// for that index. If no suitable index could be found, an error is returned
+func (ei *EntityInfo) IndexFromConditions(conditions map[string][]*Condition) (name string, key *PrimaryKey, err error) {
```

Comment:
A more interesting question is, what is an index in this context? One could argue that the primaryKey is just another index, especially since it shares the same PrimaryKey structure.